web/modules/custom/cyber_breadcrumbs/src/ViewPageBreadcrumb.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Drupal\cyber_breadcrumbs;
+
+/**
+ * @file
+ * Generates the breadcrumb trail for search page(s)
+ */
+use Drupal\Core\Breadcrumb\Breadcrumb;
+use Drupal\Core\Breadcrumb\BreadcrumbBuilderInterface;
+use Drupal\Core\Controller\TitleResolverInterface;
+use Drupal\Core\Link;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class ViewPageBreadcrumb implements BreadcrumbBuilderInterface {
+
+
+  /**
+   * Symfony\Component\HttpFoundation\RequestStack definition.
+   *
+   * @var Symfony\Component\HttpFoundation\RequestStack
+   */
+  protected $request;
+
+  /**
+   * The title resolver.
+   *
+   * @var \Drupal\Core\Controller\TitleResolverInterface
+   */
+  protected $titleResolver;
+
+  /**
+   * Class constructor.
+   */
+  public function __construct(RequestStack $request, TitleResolverInterface $title_resolver) {
+    $this->request = $request;
+    $this->titleResolver = $title_resolver;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('request_stack'),
+      $container->get('title_resolver')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function applies(RouteMatchInterface $route_match) {
+    $match = FALSE;
+    $route_name = $route_match->getRouteName();
+    $view_names = [
```

Comment:
Something of interest (rather than necessity) is that you can pass in parameters from the service container rather than baking them into the class itself. That's quite handy for long term testability and robustness. There's an example of it here https://github.com/dof-dss/nidirect-site-modules/blob/e794e62107e9bf1bc1d296a5bc5a7c9e37abe3b1/nidirect_breadcrumbs/nidirect_breadcrumbs.services.yml#L39

And wired up in https://github.com/dof-dss/nidirect-site-modules/blob/development/nidirect_breadcrumbs/src/SearchBreadcrumb.php through the constructor.